edi/models/edi_exchange_type.py

```diff
@@ -121,24 +121,36 @@ def _check_backend(self):
             if rec.backend_id.backend_type_id != rec.backend_type_id:
                 raise exceptions.UserError(_("Backend should respect backend type!"))
 
-    def _make_exchange_filename(self, exchange_record):
-        """Generate filename."""
-        pattern = self.exchange_filename_pattern
+    def _get_exchange_filename_values(self, exchange_record):
+        """
+        Get values to build the filename
+
+        Values used to build the filename,
+        based on the exchange_filename_pattern pattern
+        :param exchange_record:
+        :return: dict
+        """
         ext = self.exchange_file_ext
-        pattern = pattern + ".{ext}"
         dt = slugify(fields.Datetime.to_string(fields.Datetime.now()))
         record_name = self._get_record_name(exchange_record)
         record = exchange_record
         if exchange_record.model and exchange_record.res_id:
             record = exchange_record.record
-        return pattern.format(
-            exchange_record=exchange_record,
-            record=record,
-            record_name=record_name,
-            type=self,
-            dt=dt,
-            ext=ext,
-        )
+        return {
+            "exchange_record": exchange_record,
+            "record": record,
+            "record_name": record_name,
+            "type": self,
+            "dt": dt,
+            "ext": ext,
+        }
+
+    def _make_exchange_filename(self, exchange_record):
+        """Generate filename."""
+        pattern = self.exchange_filename_pattern
+        pattern = pattern + ".{ext}"
```

Comment:
this should come from the value defined by the user. You should not enforce ext here.

Reply:
It's just to match with this: https://github.com/OCA/edi/pull/346/files#diff-a7911660a866fa99e022eaca3b199af8df77070ec6de5f00a5021421a27a6500R8
The extension should be added into the code after the file name pattern.